tools/commands/build.py

```diff
@@ -31,9 +29,32 @@ def __str__(self):
     def clean(self):
         if os.path.exists(self.build_path):
             logger.info("cleaning {}".format(self.name))
-            os.system("xargs rm < {}/install_manifest.txt".format(self.build_path))
             shutil.rmtree(self.build_path)
```

Comment:
Do you suggest to make clean an abstract method in the base class and implement it in cmakebuilder and acbuilder (currently just removing the build directory since these builders only allow building out of tree)?

Reply:
Indeed that's what I meant.

Concretely: we will possibly want to add the build of hostapd/wpa_supplicant itself as well. AFAIK, it doesn't support out-of-tree build. So build_path will be equal to module_path.

We could handle that by copying everything to the build directory. Maybe that's indeed a better solution. In that case, we can keep the clean() implementation in the superclass like it is.

There's still a small nit: after this change, the "cleaning" message is only printed after uninstall has already been done. But I don't think it's worth fixing that.

So I'm marking this comment as resolved.